script.rb

```diff
@@ -0,0 +1,102 @@
+module Enumerable
+  # my_each
+  def my_each
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    i = 0
+    while i < to_a.length
+      yield to_a[i], index(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  # my_select
+  def my_select
+    my_arr = []
+    to_a.my_each do |num|
+      my_arr.push(num) if yield num
+    end
+    my_arr
+  end
+
+  # my_all
+  def my_all
+    new_arr = []
+    to_a.my_each do |num|
+      new_arr.push(num) if yield num
+    end
+    new_arr.length == to_a.length
+  end
+
+  # my_any
+  def my_any
+    new_arr = []
+    to_a.my_each do |num|
+      new_arr.push(num) if yield num
+    end
+    new_arr.length >= 1
+  end
```

Comment:
- [ ] should have working ```my_any?```  identical and returns the same thing as ruby's each_with_index. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))


-  when no block or argument is given returns true if at least one of the collection is not false or nil
- Your method doesn't consider the possibility when a class (String, Numeric, etc) is given as an argument ```some_enumerable.my_anyl?(String)```.